### PR TITLE
Exercise 10

### PR DESCRIPTION
**Pull request by vaibhav1sh**
Submission for Exercise-10

### CODE REVIEW

File: .env.template [hunk 1]
```diff
@@ -22,4 +22,17 @@ TRELLO_BOARD_NAME_CREATE = board_to_be_created
...
+MONGO_PASSWORD=password
+MONGO_URL=mongo-cluster-name
+
+# Mongo DB variables to connect to production database and collection. 
+TODO_DATABASE=Todo_Database
+TODO_COLLECTION=Todo_Collection
+
+# Mongo DB test variables to run integration and E2E tests
+TEST_DATABASE=Test-database
+TEST_COLLECTION=Test-collection
```
Comment (BenRamchandani) [inline .env.template]: Can you add the new variables (`OAUTH_CLIENT_ID` and `OAUTH_CLIENT_SECRET`) to here?

File: README.md [hunk 2]
```diff
@@ -197,4 +198,8 @@ getLogger("urllib3").setLevel(logging.WARNING)
 Authentication with Loggly is done through the use of customer token, which is stored in environment variable LOGGLY_TOKEN in .env file.
 
 
+## **Module-9**
 
+All the references to Trello have been replaced with Mongo database. The .env file should now include the MongoDB authentication variables, MongoDB Production Database/Collection names, and MongoDB Test Database/Collection names (for E2E tests).These are all included in .env.template file.
+
+Nothing else changes, the application should be built and deployed as per previous instructions.
```
Comment (BenRamchandani) [inline README.md]: It would be good to mention how to set GitHub OAuth up

File: todo_app/app.py [hunk 3]
```diff
@@ -24,37 +33,85 @@ def create_app():
...
 
     @app.route('/')
+    @login_required
     def index():
         app.logger.info("Start rendering template for get call")
         app.logger.info(f'Env variables are {os.environ.get("MONGO_USER_NAME")}')
         cards = fetch_all_cards("To Do", "Doing", "Done")  
         view_model_obj = ViewModel(cards) 
+        if app.config.get('LOGIN_DISABLED') or current_user.role == 'writer':
+            user_role = 'writer'
```
Comment (BenRamchandani) [inline todo_app/app.py:78]: You don't set `user_role` if this `if` expression isn't true, so it errors on line 85!

**Review by BenRamchandani — APPROVED** [review]
All looks great apart from one bug in the `index` function. You should fix that before merging, but no need to get it re-reviewed.

It's fine not to update your Heroku app, but when you deploy your app to Azure next module make sure you enable authentication!

File: todo_app/app.py (hunk 3 above)
Comment (vaibhav1sh) [reply todo_app/app.py:78]: Ah ! Thanks for the feedback - stupid error on my part

File: .env.template (hunk 1 above)
Comment (vaibhav1sh) [reply .env.template]: Done!

**Pull request merged**